lib/imap.ml

```diff
@@ -95,6 +95,7 @@ module Mutf7 = struct
     loop d e
 
   let replace s ch1 ch2 =
+    let s = Bytes.unsafe_of_string s in
```

Comment:
Rather than doing this, which is unsafe, it is better to replace line 116
```
let str = B64.encode ~pad:false (Buffer.contents buf)
```
below by
```
let str = Bytes.of_string (B64.encode ~pad:false (Buffer.contents buf))
```
so that the argument to `replace` is already of `bytes` type; you will also need to replace `Buffer.add_string b str` by `Buffer.add_bytes b str`.